src/Microsoft.DotNet.Tools.Run/Program.cs

```diff
@@ -19,52 +13,24 @@ public static int Main(string[] args)
-            app.HelpOption("-h|--help");
-
-            var framework = app.Option("-f|--framework <FRAMEWORK>", "Compile a specific framework", CommandOptionType.MultipleValue);
-            var configuration = app.Option("-c|--configuration <CONFIGURATION>", "Configuration under which to build", CommandOptionType.SingleValue);
-            var preserveTemporaryOutput = app.Option("-t|--preserve-temporary", "Keep the output's temporary directory around", CommandOptionType.NoValue);
-
-            // This is required to be an option because otherwise we can't tell if the first argument is a project or the first argument to pass to an application
-            var project = app.Option("-p|--project <PROJECT_PATH>", "The path to the project to run (defaults to the current directory). Can be a path to a project.json or a project directory.", CommandOptionType.SingleValue);
+            RunCommand runCmd = new RunCommand();
 
-            app.OnExecute(() =>
+            ArgumentSyntax.Parse(args, syntax =>
             {
-                // Locate the project and get the name and full path
-                var path = project.Value();
-                if (!string.IsNullOrEmpty(path))
-                {
-                    if (File.Exists(path) && (Path.GetExtension(path) == ".csx"))
-                    {
-                        return RunInteractive(path);
-                    }
-                }
-                else
-                {
-                    path = Directory.GetCurrentDirectory();
-                }
-
-                var contexts = ProjectContext.CreateContextForEachFramework(path);
-                ProjectContext context;
-                if (!framework.HasValue())
-                {
-                    context = contexts.First();
-                }
-                else
-                {
-                    var fx = NuGetFramework.Parse(framework.Value());
-                    context = contexts.FirstOrDefault(c => c.TargetFramework.Equals(fx));
-                }
-                return Run(context, configuration.Value() ?? Constants.DefaultConfiguration, app.RemainingArguments, preserveTemporaryOutput.HasValue());
+                syntax.HandleErrors = false;
+                syntax.DefineOption("f|framework", ref runCmd.Framework, "Compile a specific framework");
+                syntax.DefineOption("c|configuration", ref runCmd.Configuration, "Configuration under which to build");
+                syntax.DefineOption("t|preserve-temporary", ref runCmd.PreserveTemporary, "Keep the output's temporary directory around");
+                syntax.DefineOption("p|project", ref runCmd.Project, "The path to the project to run (defaults to the current directory). Can be a path to a project.json or a project directory");
+
+                // TODO: this is not supporting args which can be switches (i.e. --test)
+                // TODO: we need to make a change in System.CommandLine or parse args ourselves.
```

Comment:
Should we create an issue for this? I think this is an important case in the some of the other commands.